### ISSUE TITLE
Provide a way for users to use requests.Session

### ISSUE THREAD

**Issue by jarshwah** [SiftScience/sift-python#64]
We're trying to use [connection pooling](http://docs.python-requests.org/en/master/user/advanced/#session-objects) which reduces our latency from 800ms to 200ms but we're finding it very difficult to "swap out" the `requests.get` for a `session.get` due to the design of the class.

Could you please add a method hook that we could override in a subclass that returns the requests/session object? Something like this would be perfect:

```
def get_requester(self):
    return requests
```

Then in a subclass, we could override that method like so:

```

def __init__(self, *args, **kwargs):
    self.session = requests.Session( .. )
    super(Class, self).__init__(*args, **kwargs)

def get_requester(self):
    return self.session
```

And then each of the methods that needs to fire off a web request would use get_requester rather than using `requests` directly.

**Comment by quanpod** [comment]
Thanks for filing @jarshwah - I'll have someone from the team take a look.  (cc @garylee1 and @ehrmann for triage).  Mind sending me an email at quan@siftscience.com?  I'm assuming you're seeing this because you're based in Oz, but would love more details.

**Comment by jarshwah** [comment]
Correct, I am from Australia. Happy to send an email also - will do so from my company address (kogan).

Another direction that you could go with that might be better for all of your users is to switch to using session exclusively, but allow users to pass a session in.

```
def __init__(self, ..., session=None):
    ...
    self.session = session or requests.Session()
    self.session.auth = ( .. )

def track(...):
    ...
    self.session.post( ... )
```

In this way you get connection pooling by default if using the same instance of the client. In our case, we're going to maintain a session in a threadlocal so that we can maintain connection pooling across multiple web requests, and then we can pass that into the constructor.

**Comment by ehrmann** [comment]
Fixed in f89de0a
